ambry-tools/src/main/java/com.github.ambry/tools/admin/ConcurrencyTestTool.java

```diff
@@ -474,6 +505,81 @@ public void onCompletion(Void result, Exception exception) {
   }
 
   /**
+   * Class to assist in collecting and reporting metrics
+   */
+  class MetricsCollector {
+    private String operation;
+    private final long measurementIntervalInSecs;
+    private ArrayList<Long> requestLatencies;
+    private AtomicLong timePassedInMs = new AtomicLong(0);
+    private AtomicLong numberOfOperations = new AtomicLong(0);
+    private AtomicLong maxLatencyInMs = new AtomicLong(0);
+    private AtomicLong minLatencyInMs = new AtomicLong(Long.MAX_VALUE);
+    private AtomicLong totalLatencyInMs = new AtomicLong(0);
+
+    MetricsCollector(long measurementIntervalInSecs, String operation) {
+      this.measurementIntervalInSecs = measurementIntervalInSecs;
+      requestLatencies = new ArrayList<Long>();
```

Comment:
Also, codahale seems to have a class called `SlidingTimeWindowReservoir`, but it might be overkill for this use case.